File: examples/ruby/url.rb
#!/usr/bin/ruby

require 'zypp'
include Zypp

a = Url.new("http://www.suse.de/download")
puts "a = #{a}"
puts "    #{a.get_scheme} #{a.get_host} #{a.get_path_data}"

b = Url.new(a)
b.set_host("download.novell.com")
puts "b = #{b}"

c = a                   # oops
c.set_path_data("/repository")
puts "c = #{c}"

puts
puts "a = #{a}"         # oops
puts "b = #{b}"
puts "c = #{c}"

File: examples/python/url.py
#!/usr/bin/python

from zypp import Url

a = Url("http://www.suse.de/download")
print "a = %s" % a
print "    %s %s %s" % (a.getScheme(), a.getHost(), a.getPathData())

b = Url(a)
b.setHost("download.novell.com")
print "b = %s" % b

c = a                   # oops
c.setPathData("/repository")
print "c = %s" % c

print
print "a = %s" % a      # oops
print "b = %s" % b
print "c = %s" % c

